lib/class-controller.php

```diff
@@ -386,6 +386,17 @@ public function load_adapter( ?Adapter $adapter = null ): void {
 		}
 	}
 
+	/**
+	 * Directly query Elasticsearch.
+	 *
+	 * @param array $es_args The query to send to Elasticsearch.
+	 *
+	 * @return array The response from Elasticsearch.
+	 */
+	public function search( array $es_args ): array {
```

Comment:
Nitpicky but I'd prefer this be `query` instead of `search`

Reply:
I'm more torn on this, personally. In ES, both words have meaning, so we have some language crossover. In ES, `search` is the endpoint that this is going to POST to, and `query` is part of the DSL, but `$es_args` may not even include a query, if it's just using aggs. With all that in mind, I lean toward `search()` being the method name, so it more closely reflects the language ES uses. It's a tough one, and I've probably even argued in favor of naming something like this `query()` elsewhere!

Reply:
Sold! `search` it is.

Reply:
@mboynes makes me think of something potentially worth confirming here: that SP and EP's `search()` methods  are doing the same thing with their respective `$es_args`.

Reply:
what i mean to ask is whether or not we need a transformer to ensure that the value for `$es_args` are consistent between the two adapters.

Reply:
I've confirmed that this is working for both SP and EP, but one ting worth noting is that EP for some reason doesn't like the date decay clause in the example in the function score above. aside from that, I am able to reproduce the more_like_this query (among other custom queries in both adapters with little effort.